### ISSUE TITLE
Provide a way to get feedback on compiler error rates

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#45857]
As per https://github.com/rust-lang/rust/pull/45452#issuecomment-339329308, and [other conversations](https://internals.rust-lang.org/search?q=telemetry) that have been had over time, there's a feeling that we should have ways of identifying what `rustc` features are being used/encountered. This can range from full featured telemetry support in `rustc`/`cargo`, to a simple webapp accepting people to post compiler output. Right now the only ways we receive feedback for confusing output are:

 - IRC/gitter
 - this issue tracker
 - stackoverflow
 - /r/rust

If we added telemetry support on rustc, it would also _have_ to be opt-in, and it could start with being a simple flag passed to `rustc` to send a post request with the text output of the compiler. These reports could then be aggregated by type.

**Comment by kennytm** [comment]
I think we better keep `rustc` untouched, and telemetry is better belong on `cargo`:

* the "flag" can be part of `~/.cargo/config` so you don't need to opt-in everytime
* the "flag" can be configured when you install `rustup` the first time
* `rustc` remains free of any network code
* `cargo` already needs to connect to the internet (outside of airplane mode)

**Comment by nikomatsakis** [comment]
@kennytm those sound like solid arguments to me

**Comment by bluss** [comment]
Rustup already has a config flag for telemetry. https://github.com/rust-lang-nursery/rustup.rs/issues/341

**Comment by kennytm** [comment]
Oh in this case the entire thing can be implemented on top of rustup's `rustc`/`cargo` wrapper already?

**Comment by estebank** [comment]
cc https://internals.rust-lang.org/t/pre-pre-rfc-automatically-generate-bug-reports/7736

**Comment by steveklabnik** [comment]
I don't believe that we've done anything in this area in the past few years.

**Comment by estebank** [comment]
I've formalized some of my thoughts on the matter: https://estebank.github.io/rustc-metrics.html
